test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -9271,14 +9262,12 @@ def create_vnodes(self, name=None, attrib=None, num=1, mom=None,
         # only expect if vnodes were added rather than the nat vnode modified
         if expect and num > 0:
             attrs = {}
-            n = num + len(existing_vnodes)
-            if len(existing_vnodes) > 0:
-                n = n - 1
-            for k, v in attrib.items():
-                attrs[str(k) + '=' + str(self.utils.decode_value(v))] = n
-            attrs['state=free'] = n
-            attrs['resources_available.host=' + natvnode] = n
-            rv = self.expect(VNODE, attrs, attrop=PTL_AND)
+            attrs.update({'state': 'free'})
+            attrs.update(attrib)
+            for vn in new_vnodelist:
+                rv = self.expect(VNODE, attrs, id=vn)
+                if not rv:
+                    break
```

Comment:
No need to check the return value of rv.  If the expect() fails, it will raise an exception.  Now you might consider putting it in a try/except block.